reconcile/utils/saasherder.py

```diff
@@ -226,6 +233,22 @@ def _check_saas_file_env_combo_unique(self, saas_file_name, env_name):
         else:
             self.tkn_unique_pipelineruns[tkn_name] = tkn_long_name
 
+    def _validate_upstream_not_used_with_commit_sha(
+            self,
+            saas_file_name: str,
+            resource_template_name: str,
+            target: dict,
```

Comment:
We'd usually mark `target` as `Mapping` - maybe even `Mapping[str, Any]`, but this module is already hard enough to make a fuss here.

Reply:
:+1: